gobblin-service/src/test/java/org/apache/gobblin/service/modules/flow/MultiHopFlowCompilerTest.java

```diff
@@ -670,6 +670,28 @@ public void testUnresolvedFlow() throws Exception {
     spec.getCompilationErrors().stream().anyMatch(s -> s.contains(AzkabanProjectConfig.USER_TO_PROXY));
   }
 
+  @Test (dependsOnMethods = "testUnresolvedFlow")
+  public void testMissingSourceNodeError() throws Exception {
+    FlowSpec spec = createFlowSpec("flow/flow5.conf", "HDFS-NULL", "HDFS-3", false, false);
+
+    Dag<JobExecutionPlan> dag = specCompiler.compileFlow(spec);
+
+    Assert.assertEquals(dag, null);
+    Assert.assertEquals(spec.getCompilationErrors().size(), 1);
+    spec.getCompilationErrors().stream().anyMatch(s -> s.contains("Flowgraph does not have a node with id"));
+  }
+
+  @Test (dependsOnMethods = "testMissingSourceNodeError")
+  public void testMissingDestinationNodeError() throws Exception {
```

Comment:
Am I missing something, or shouldn't this require validation on the `List<DataNode> destNodes` object? Right now it's only validating the `DataNode sourceNode` right?

Reply:
Ah my bad I added that way too fast, good catch

Reply:
@jack-moseley you're right, it reports the error correctly on local which is what I tested, but when we add our dmas authorizer it runs into the nullpointer exception before it does another check when calculating the path, I'll add the error check now.